skimage/measure/_find_contours_cy.pyx

```diff
@@ -93,13 +103,65 @@ def iterate_and_store(double[:, :] array,
             coords[0] += 1
             coords[1] = 0
 
-
         square_case = 0
         if (ul > level): square_case += 1
         if (ur > level): square_case += 2
         if (ll > level): square_case += 4
         if (lr > level): square_case += 8
 
+        # We need to handle missing data.
+        # (This could either be in the form of NaNs present in the input array,
+        # or could be the result of masking.)
+        # Start by counting the number of missing data values.
+        nan_count = 0
+        if npy_isnan(ul): nan_count += 1
+        if npy_isnan(ur): nan_count += 1
+        if npy_isnan(ll): nan_count += 1
+        if npy_isnan(lr): nan_count += 1
+
+        if nan_count > 1:
+            # If a square has 2 or more missing values, we cannot correctly
+            # infer the presence of any contour line segments within it;
+            # so just move to the next square.
+            continue
+        elif nan_count == 1:
+            # There is (up to symmetry) one square arrangement containing a
+            # missing value in which we can unambiguously draw an isoline
+            # segment:
+            # The arrangement +-
+            #                 -x (where x denotes a missing value)
+            # should have the same contour line as case 1.
+            # After symmetry, there are 8 cases, which are enumerated here.
```

Comment:
Is this true though? The way I understand this, the missing value above might either be + or - corresponding to the cases 1 and 9. So we actually don't know what the correct iso line might be.

Also could you elaborate what you mean with "after symmetry"?

Reply:
By symmetry, I mean the fact that cases 1,2,4,8 are just rotations of each other, and furthermore these are the same as 14,13,11,7 with + and - swapped (you might think of this as reflection in the Z coordinate). Whatever we decide is the correct treatment for one of these cases can be transferred to the others by applying the correct symmetry.

My thinking was that the outputs in case 1 (top-left segment) and case 9 (top-left and bottom-right segments) both include the top-left line segment, so we could unambiguously output this segment; but I see now that I might have been too hasty: 9 is one of the ambiguous "saddle points", so it could instead output top-right and bottom-left.

I think there are probably still some cases where this really isn't ambiguous if we take `vertex_connect_high` in to account; but I also might just be trying to get too clever here - since contours are going to be cut off by the mask edge anyway, losing a single pixel probably isn't a big deal.